### PR TITLE
Github-issue#1048 : s3-sink integration test implementation.

### PR DESCRIPTION
**Pull request by deepaksahu562**
### Description

- Implementation of s3-sink integration test.

 
### Issues Resolved
GitHub-issue #1048 
 
### Check List
- [x] New functionality includes testing.
- [x] New functionality has been documented.
  - [x] New functionality has javadoc added
- [x] Commits are signed with a real name per the DCO

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/data-prepper/blob/main/CONTRIBUTING.md).

### CODE REVIEW

File: data-prepper-plugins/s3-sink/src/integrationTest/java/org/opensearch/dataprepper/plugins/sink/S3SinkServiceIT.java [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+    }
+
+    @Test
+    void verify_flushed_records_into_s3_bucket() {
+
+        int s3ObjectCountBeforeIngest = gets3ObjectCount();
+        S3SinkService s3SinkService = createObjectUnderTest();
+        s3SinkService.output(setEventQueue());
+        int s3ObjectCountAfterIngest = gets3ObjectCount();
+        assertThat(s3ObjectCountAfterIngest, greaterThan(s3ObjectCountBeforeIngest));
```
Comment (kkondaka) [inline data-prepper-plugins/s3-sink/src/integrationTest/java/org/opensearch/dataprepper/plugins/sink/S3SinkServiceIT.java:94]: This looks unnecessary.

Comment (deepaksahu562) [reply data-prepper-plugins/s3-sink/src/integrationTest/java/org/opensearch/dataprepper/plugins/sink/S3SinkServiceIT.java]: Resolved.

**Review by ashoktelukuntla — DISMISSED** [review]
Look good

**Review by kkondaka — DISMISSED** [review]
Looks good.

File: data-prepper-plugins/s3-sink/src/integrationTest/java/org/opensearch/dataprepper/plugins/sink/S3SinkServiceIT.java [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+    private int gets3ObjectCount() {
+        int s3ObjectCount = 0;
+        try {
+            ListObjectsRequest listObjects = ListObjectsRequest.builder().bucket(bucketName).build();
+            ListObjectsResponse res = s3Client.listObjects(listObjects);
+            List<S3Object> objects = res.contents();
+            s3ObjectCount = objects.size();
+        } catch (S3Exception e) {
+            LOG.error(e.awsErrorDetails().errorMessage());
+            System.exit(1);
```
Comment (dlvenable) [inline data-prepper-plugins/s3-sink/src/integrationTest/java/org/opensearch/dataprepper/plugins/sink/S3SinkServiceIT.java:110]: This is going to fail the whole test suite and will prevent accurate reporting of failures.

Just let the exception be thrown here. Remove the try-catch block entirely. JUnit will handle it.

File: data-prepper-plugins/s3-sink/src/integrationTest/java/org/opensearch/dataprepper/plugins/sink/S3SinkServiceIT.java [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+
+    @Test
+    void verify_flushed_records_into_s3_bucket() {
+
+        int s3ObjectCountBeforeIngest = gets3ObjectCount();
+        S3SinkService s3SinkService = createObjectUnderTest();
+        s3SinkService.output(setEventQueue());
+        int s3ObjectCountAfterIngest = gets3ObjectCount();
+        assertThat(s3ObjectCountAfterIngest, equalTo(s3ObjectCountBeforeIngest + 1));
+    }
```
Comment (dlvenable) [inline data-prepper-plugins/s3-sink/src/integrationTest/java/org/opensearch/dataprepper/plugins/sink/S3SinkServiceIT.java:95]: This test should actually load the S3 object and verify the data is correct.

**Review by dlvenable — CHANGES_REQUESTED** [review]

File: data-prepper-plugins/s3-sink/src/integrationTest/java/org/opensearch/dataprepper/plugins/sink/S3SinkServiceIT.java (hunk 2 above)
Comment (deepaksahu562) [reply data-prepper-plugins/s3-sink/src/integrationTest/java/org/opensearch/dataprepper/plugins/sink/S3SinkServiceIT.java]: Removed the try-catch block.

File: data-prepper-plugins/s3-sink/src/integrationTest/java/org/opensearch/dataprepper/plugins/sink/S3SinkServiceIT.java (hunk 3 above)
Comment (deepaksahu562) [reply data-prepper-plugins/s3-sink/src/integrationTest/java/org/opensearch/dataprepper/plugins/sink/S3SinkServiceIT.java:121]: Resolved.

**Comment by deepaksahu562** [comment]
@ashoktelukuntla \ @dlvenable 
Addressed the change request, would you please review and merge it.

**Review by kkondaka — APPROVED** [review]
Looks good.

**Review by ashoktelukuntla — APPROVED** [review]
Looks good

**Review by dlvenable — APPROVED** [review]

**Pull request merged**